packages/elements/src/Header/HeaderBackButton.tsx

```diff
@@ -1,19 +1,79 @@
-import { useTheme } from '@react-navigation/native';
+import { Link, useLinkTools, useTheme } from '@react-navigation/native';
 import * as React from 'react';
 import {
   Animated,
   I18nManager,
   Image,
   LayoutChangeEvent,
   Platform,
+  StyleProp,
   StyleSheet,
   View,
+  ViewStyle,
 } from 'react-native';
 
 import MaskedView from '../MaskedView';
 import PlatformPressable from '../PlatformPressable';
 import type { HeaderBackButtonProps } from '../types';
 
+const LinkPressable = ({
+  href,
+  children,
+  style,
+  onPress,
+  onLongPress,
+  onPressIn,
+  onPressOut,
+  accessibilityRole,
+  ...rest
+}: Omit<React.ComponentProps<typeof PlatformPressable>, 'style'> & {
+  style: StyleProp<ViewStyle>;
+} & {
+  href?: string;
+  children: React.ReactNode;
+  onPress?: () => void;
+}) => {
+  const { buildAction } = useLinkTools();
+  if (Platform.OS === 'web' && href) {
+    // React Native Web doesn't forward `onClick` if we use `TouchableWithoutFeedback`.
+    // We need to use `onClick` to be able to prevent default browser handling of links.
+    return (
+      <Link
+        {...rest}
+        href={href}
+        action={buildAction(href)}
```

Comment:
We should add `e.defaultPrevented` check inside `Link`'s `onPress` so that this `action` prop is not needed. Otherwise I think it'll dispatch 2 actions, one for this and one for the `onPress` prop.